File: algo-c-to-_/examples/mccarthy_.py
#
#	from src/mccarthy.c
#
#	int McCarthy(int)	to	mccarthy91
#

from mccarthy import mccarthy91
import sys

def _t_mccarthy91(l, r):
    for i in range(l, r+1):
        t = mccarthy91(i)
        if t != 91:
            print("ERROR: {0} {1}".format(i, t))
            sys.exit()
    print("mccarthy91 seems to be 91 in {0} to {1}".format(l, r))

_t_mccarthy91(-(1 << 10), 100)

print("... and in 101 to 110 are:")

for i in range(101, 111):
    sys.stdout.write("{0:4d}:{1}".format(i, mccarthy91(i)))
print()

File: porting-from-algo-c/examples/mccarthy.lua
--
--	from src/mccarthy.c
--
--	int McCarthy(int)	to	mccarthy91
--

local mccarthy91 = require 'mccarthy'.mccarthy91

local function _t_mccarthy91(l, r)
	for i=l,r do
		local t = mccarthy91(i)
		if t ~= 91 then
			print(("ERROR: %d %d"):format(i, t))
			os.exit()
		end
	end
	print(("mccarthy91 seems to be 91 in %d to %d"):format(l, r))
end

do
	_t_mccarthy91(-(1 << 10), 100)

	print("... and in 101 to 110 are:")

	for i=101,110 do
		io.write(("%4d:%d"):format(i, mccarthy91(i)))
	end
	print()
end
